mozilla-services/consvc-shepherd
# [SHEP-79] - Record BQ Sync Runs

## References

[SHEP-79](https://mozilla-hub.atlassian.net/browse/SHEP-79)

## Problem Statement

We need to add a new model to our database that will record daily runs of our script to fetch data from BQ so we can easily check that the script is working as intended.

## Proposed Changes

- Include new `BQSyncStatus` model/migration to save the status of `sync_bq_data` script
- Modify `BoostrSyncStatus` model to ensure correct pluralization
- Modify `sync_bq_data` to fail if it can't query for a specific date 
- Unit test enhancements

## Verification Steps

1. Apply the migrations
2. Run shepherd and exec with `make debug`
3. Run the script with `python manage.py sync_bq_data --project_id "moz-fx-ads-nonprod"` and check the table for the status
4. Try to break the script to ensure failures are recorded as well. For example, provide a future date that we won't have data for such as `2025-10-07` or modify the BQ query to query something that doesn't exist, etc.
5. Run the tests and make sure that they pass

[SHEP-79]: https://mozilla-hub.atlassian.net/browse/SHEP-79?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
